pkg/controllers/monitoring-stack/components.go

```diff
@@ -427,11 +475,7 @@ func newPrometheus(
 		prometheusSelector = &metav1.LabelSelector{}
 	}
 
-	// create a default config if user hasn't specified one
-	config := ms.Spec.PrometheusConfig
-	if config == nil {
-		config = &stack.PrometheusConfig{}
-	}
+	config := applyDefaultConfig(ms.Spec.PrometheusConfig)
```

Comment:
Do we need to apply the default explicitly here? Doesn't the default in the CRD manifest take care of that?

Reply:
I think it can if we change `PrometheusConfig *PrometheusConfig` in MonitoringStackSpec to `PrometheusConfig PrometheusConfig` since it  `defaults` to `nil` at the moment.  

I don't have a strong opinion about the implementation. What do you feel?

Reply:
Good catch! I have modified the spec to apply defaults. My previous attempt didn't work since I had to be explicit at 2 different places - `config`, and `replicas`